File: cwclock-mobile/src/components/ToggleRow.js
import React from "react";
import { StyleSheet, Switch, Text, View } from "react-native";
import theme from "../theme";

// ToggleRow is a labeled Switch row - used for the All day / Half day
// toggles (EditRecordScreen), which disable each other exactly like the web
// app's TaskComponent.jsx/ReportEntryRow.jsx (ai-instruct-104).
const ToggleRow = ({ label, value, onValueChange, disabled, style }) => (
  <View style={[styles.row, style]}>
    <Text style={[styles.label, disabled && styles.labelDisabled]}>{label}</Text>
    <Switch
      value={value}
      onValueChange={onValueChange}
      disabled={disabled}
      trackColor={{ true: theme.color.primary }}
    />
  </View>
);

const styles = StyleSheet.create({
  row: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingVertical: theme.spacing(1),
  },
  label: {
    fontSize: 16,
    color: theme.color.text,
  },
  labelDisabled: {
    color: theme.color.textMuted,
  },
});

export default ToggleRow;

File: cwclock-mobile/lib/widgets/toggle_row.dart
import 'package:flutter/material.dart';

import '../theme.dart';

/// A labeled Switch row - ported from src/components/ToggleRow.js. Used for
/// the All day / Half day toggles (EditRecordScreen), which uncheck each
/// other instead of disabling.
class AppToggleRow extends StatelessWidget {
  final String label;
  final bool value;
  final ValueChanged<bool> onChanged;
  final bool disabled;

  const AppToggleRow({
    super.key,
    required this.label,
    required this.value,
    required this.onChanged,
    this.disabled = false,
  });

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: EdgeInsets.symmetric(vertical: AppSpacing.of(1)),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            label,
            style: TextStyle(fontSize: 16, color: disabled ? AppColors.of(context).textMuted : AppColors.of(context).text),
          ),
          Switch(
            value: value,
            onChanged: disabled ? null : onChanged,
            activeTrackColor: AppColors.of(context).primary,
          ),
        ],
      ),
    );
  }
}
